cleedpy/rfactor.py

```diff
@@ -118,3 +116,71 @@ def energy_step(energies):
     """
 
     return energies[1:] - energies[:-1]
+
+
+def split_in_pairs(exp_iv, theo_iv):
+    """This function takes a complete datasets with a experimental and theoretical iv curves splits them them in pairs corresponding to the same index."""
+
+    beam_indices = np.unique(np.vstack([exp_iv, theo_iv])[:, [0, 1]], axis=0)
+
+    for indx1, indx2 in beam_indices:
+        exp_curve = exp_iv[(exp_iv[:, 0] == indx1) & (exp_iv[:, 1] == indx2)][
+            :, [-2, -1]
+        ]
+        theo_curve = theo_iv[(theo_iv[:, 0] == indx1) & (theo_iv[:, 1] == indx2)][
+            :, [-2, -1]
+        ]
+        yield exp_curve, theo_curve
+
+
+def find_common_x_axis(x1, x2):
+    min_x = max(np.min(x1), np.min(x2))
+    max_x = min(np.max(x1), np.max(x2))
+
+    x1 = x1[(x1 >= min_x) & (x1 <= max_x)]
+    x2 = x2[(x2 >= min_x) & (x2 <= max_x)]
+
+    return np.unique(np.sort(np.concatenate([x2, x1])))
+
+
+def compute_rfactor(experimental_iv, theoretical_iv, shift=0.0, rfactor_type="r2"):
+
+    r_tot = 0.0
+    rfactor = {
+        "r2": r2_factor,
+        "pendry": rp_factor,
+    }
+
+    # Looping over the pairs of experimental and theoretical curves corresponding to the same index.
+    for exp_curve, theo_curve in split_in_pairs(experimental_iv, theoretical_iv):
+        if len(exp_curve) < 1 or len(theo_curve) < 1:
+            continue
+        # Perform Lorentzian smoothing.
+        exp_curve = lorentzian_smoothing(exp_curve)
+        theo_curve = lorentzian_smoothing(exp_curve)
```

Comment:
```suggestion
        theo_curve = lorentzian_smoothing(theo_curve)
```

Reply:
fixed in https://github.com/empa-scientific-it/cleedpy/pull/56/commits/8606c12130fa36e4a671bae1f3ef6945fab5eb94